include/wtlgo/Network.hpp

```diff
@@ -23,6 +25,9 @@ class Network {
     
     bool download(const std::string& url, const std::string& save_as = "") const;
     
+    void set_logger(std::shared_ptr<ILogger> obj);
+    std::shared_ptr<ILogger> get_logger();
```

Comment:
Please, use `const` for methods that doesn't change the state.
```suggestion
    std::shared_ptr<ILogger> get_logger() const;
```